src/core/projectinfo.cpp

```diff
@@ -355,18 +390,17 @@ void ProjectInfo::restoreSettings( QString &projectFilePath, QgsProject *project
     mapCanvas->mapSettings()->setIsTemporal( isTemporal );
   }
 
-  settings.beginGroup( QStringLiteral( "/qgis/projectInfo/%1/layerstyles" ).arg( projectFilePath ) );
-  const QStringList ids = settings.allKeys();
+  settings.beginGroup( QStringLiteral( "/qgis/projectInfo/%1/layerStyles" ).arg( projectFilePath ) );
+  QStringList ids = settings.childGroups();
```

Comment:
```suggestion
  const QStringList ids = settings.childGroups();
```

Reply:
@m-kuhn , I'm recycling this below when fetching snapping settings child groups , hence why it's not const.